packages/core/src/utils/components.ts

```diff
@@ -1,17 +1,26 @@
 import { CONTENTFUL_COMPONENTS } from '../constants';
 
-export const isContentfulStructureComponent = (componentId?: string) =>
-  [
-    CONTENTFUL_COMPONENTS.section.id,
-    CONTENTFUL_COMPONENTS.columns.id,
-    CONTENTFUL_COMPONENTS.container.id,
-    CONTENTFUL_COMPONENTS.singleColumn.id,
-  ].includes(componentId ?? '');
+const allStructureComponents = [
+  CONTENTFUL_COMPONENTS.section.id,
+  CONTENTFUL_COMPONENTS.columns.id,
+  CONTENTFUL_COMPONENTS.container.id,
+  CONTENTFUL_COMPONENTS.singleColumn.id,
+];
+
+export const isContentfulStructureComponent = (
+  componentId?: string,
+  excludeStructureIds?: string[],
+) => {
+  const searchIds = excludeStructureIds
+    ? allStructureComponents.filter((id) => !excludeStructureIds?.includes(id))
```

Comment:
Just food for thought, I'd imagine there would only ever be a few dozen or so max structure components to lookup/exclude based on how many times a user will put on the canvas. However, if there's ever a scenario or edge case a user somehow uses thousands of components we should consider using a set for a faster lookup O(1) over the O(n) we have here.

Reply:
That's a very good point